_posts/2017-11-08-cost-of-drupal-7.md

```diff
@@ -0,0 +1,194 @@
+
+<!-- @todo: how best to cite an author with a quote? I don't do it consistently   -->
+
+### WIIFM? Performance.
+
+If you take performance seriously, [which you should](https://blog.kissmetrics.com/loading-time/), there's a lot to like about Drupal 8. Sticking with the theme of sophistication, Drupal 8 provides a much more granular ability to cache specific components than its predecessors. And as we know in the high-performance web world, [cache](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) [is](http://blog.catchpoint.com/2017/03/13/caching-web-performance/) [king](https://www.digitalocean.com/community/tutorials/web-caching-basics-terminology-http-headers-and-caching-strategies). When Drupal 8 first came out, a leading Acquia engineer showed some [mixed results](https://www.jeffgeerling.com/blog/2016/yes-drupal-8-slower-drupal-7-heres-why) on Drupal 8 performance. The heavier codebase invariably means having to swim upstream to make it outperform the lighter codebase in Drupal 7, but I'm happy to say the architects had their flippers on when working through these challenges. Take these two fundamental points:
+
+<!-- @todo: Did he comment back yet on my question -->
+
+1. Regardless of how fast the underlying code executes, what matters to users is perceived performance, i.e. how long they have to wait to interact with the page. Perceived delay has been drastically reduced by an experimental-turned-core module (more on that later) called [BigPipe](https://www.drupal.org/docs/8/core/modules/bigpipe/overview). BigPipe loads components of a page in the order in which a user is expected to interact with it while delivering more expensive components as they're available. This breaks with the tradition of all-or-nothing webpages served by Drupal that were either in the cache or not, lending itself to a Facebook-like experience, which is where [BigPipe came from](https://www.facebook.com/notes/facebook-engineering/bigpipe-pipelining-web-pages-for-high-performance/389414033919/).<br><br>
+ ![GIF of Drupal 8 BigPipe Video]({{ site.url }}/assets/img/blog/BigPipe In Drupal 8.gif)<br>
+  <span class="caption"> [Slower video here](https://www.youtube.com/watch?time_continue=41&v=JwzX0Qv6u3A ) </span>
+2. Modern performance tactics derive the largest gains from _outside_ of the application leveraging services like a Content Delivery Network (CDN), and/or a web application accelerator, [like Varnish](https://varnish-cache.org/intro/index.html#intro), to serve up resources to anonymous traffic (users _not_ signed-in) as quickly as possible. For most sites, anonymous traffic comprises a majority of overall traffic. Traditionally, there have been limitations to improving performance for authenticated traffic, and that's where Drupal 8 shines. With BigPipe and a more granular caching system, Drupal 8 can substantially outperform Drupal 7's authenticated user experience, so it's a win-win.
```

Comment:
I suggest changing `of the application leveraging services` to `of the application by leveraging services`

Reply:
Your reference to `authenticated user experience` in this line

> With BigPipe and a more granular caching system, Drupal 8 can substantially outperform Drupal 7's authenticated user experience

Seems somewhat out of place. Up to this point, you're talking about features that benefit anonymous users and then there's this line referencing the authenticated user experience. I don't know about Big Pipe, but the Dynamic Page Cache is the Drupal 8 feature that impacts authenticated users. [Here's Drupal.org's reference](https://www.drupal.org/docs/8/core/modules/dynamic-page-cache/overview). But you may want to say somthing like:

> Drupal 8 also comes with the Dynamic Page Cache module, which caches pages minus the personalized parts, which improves performance for all users (both anonymous & authenticated), something that wasn't possible in Drupal 7.

I'm pretty confident that some CDN's can be configured to work with the Dynamic Page Cache as well - again, benefiting all users.